features/CTX000_Presentation-colours-and-textures.feature

```diff
@@ -0,0 +1,22 @@
+@implementer-agreement
+@CTX
+@version1
+@E00020
+
+Feature: CTX000 - Presentation colours and textures
+    The rule verifies the presence of colour, texture and other presentation appearance information to objects.
+    https://ifc43-docs.standards.buildingsmart.org/IFC/RELEASE/IFC4x3/HTML/concepts/Product_Shape/Product_Geometry_Colour/content.html
+
+    Scenario Outline: Check for activation
+
+        Given an <entity type>
+        Given its attribute <attribute> 
+
+        Then The IFC model contains information on the selected functional part
+
+        Examples:
+            | entity type | attribute |
+            | IfcSolidModel | StyledByItem |
```

Comment:
I'm a little bit in doubt here. Just IfcSolidModel is definitely not enough, but introducing everything based on IfcRepresentationItem is also a bit silly, because you'd get a green box for example if you color an IfcDirection which is ridiculous, but that's because the IFC schema is ridiculous and who are we to say a direction cannot be colored.

So we can either try to come up with a comprehensive list including things like IfcFace, IfcFaceBasedSurfaceModel, IfcConnectedFaceSet, IfcShellBasedSurfaceModel, IfcCurve (yes, curves can be colored too), IfcSurface

Or just accept the general category IfcRepresentationItem. Colouring direction vectors is not something that frequently happens.

Reply:
I would say use the general category IfcRepresentationItem.  The goal of this rule is to report on models that support presentation features and we should accommodate all types that are valid in the schema, regardless of how strange it might seem.

Reply:
Implemented [here](https://github.com/buildingSMART/ifc-gherkin-rules/pull/312/commits/948bdeefc3407c0d9d9ce0c45382578cbbe8cff2)
```
    Scenario Outline: Check for activation - RepresenationItem attributes

        Given an IfcRepresentationItem
        Given Its attribute <attribute>

        Then The IFC model contains information on the selected functional part

        Examples:
            | attribute | 
            | StyledByItem |
            | HasColours | 
    
```